src/error.rs

```diff
@@ -1,64 +1,55 @@
 use crate::shares::{MAX_BITS, MIN_BITS};
 
-#[derive(Debug)]
+#[derive(Debug, thiserror::Error)]
```

Comment:
I would postpone this change a bit, until the moment we know how integrated this crate will be into Signer. Signer's errors are made to be isolated between hot and cold code like this: https://github.com/paritytech/parity-signer/blob/master/rust/definitions/src/error.rs

Figuring out if - and how - we can implement this with this macro will take some time. The decision should be made before crate publishing though - but that's delayed by zeroize issue anyway.

Reply:
I can't follow the reasoning here, it's sugar, it doesn't add magic beans and avoids the manual impl of `std::fmt::Display` and `std::error::Error`. Since publishing is not a immediate concern, I'd rather merge and adjust in the future than let the PR hang in limbo.

Reply:
You are right, we can indeed merge it and revert if `Show` is needed after all.

Reply:
Note that the `std::fmt::Display` implies the implementer can always obtain a `String` by calling `ToString` https://doc.rust-lang.org/std/string/trait.ToString.html#impl-ToString-6